badius-osic-green-tribes/src/test/java/com/greenfox/exam/badiusosicgreentribes/transaction/handler/ProductionHandlerTest.java

```diff
@@ -0,0 +1,140 @@
+package com.greenfox.exam.badiusosicgreentribes.transaction.handler;
+
+import com.greenfox.exam.badiusosicgreentribes.domain.battle.Army;
+import com.greenfox.exam.badiusosicgreentribes.domain.battle.Troop;
+import com.greenfox.exam.badiusosicgreentribes.domain.battle.Unit;
+import com.greenfox.exam.badiusosicgreentribes.domain.battle.UnitType;
+import com.greenfox.exam.badiusosicgreentribes.domain.kingdom.BuildingType;
+import com.greenfox.exam.badiusosicgreentribes.domain.kingdom.Kingdom;
+import com.greenfox.exam.badiusosicgreentribes.domain.kingdom.Storage;
+import com.greenfox.exam.badiusosicgreentribes.domain.transaction.Production;
+import com.greenfox.exam.badiusosicgreentribes.domain.transaction.ProductionUnitType;
+import com.greenfox.exam.badiusosicgreentribes.domain.transaction.TransactionState;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+class ProductionHandlerTest {
+
+    ProductionHandler handler = new ProductionHandler();
+
+    @Test
+    void confirm_CompletedTransaction() {
+        Kingdom kingdom = new Kingdom();
+        Storage storage = Storage.builder()
+                .gold(100)
+                .food(50)
+                .build();
+        kingdom.setStorage(storage);
```

Comment:
Ez közös két tesztben is. Szerintem kiemelhető egy private method-ba az osztály alján ( pl.: createKingdom() )